src/test/unit/algorithms/SpatialPoolerTest.cpp

```diff
@@ -2222,6 +2222,23 @@ namespace {
     EXPECT_EQ(0, countNonzero(activeColumns));
   }
 
+
+  TEST(SpatialPoolerTest, testSameOutputForSameInputNoLearningNoBoosting)
+  {
+    const UInt inputSize = 10;
+    const UInt nColumns = 20;
+    SpatialPooler sp;
+    sp.initialize({inputSize}, {nColumns});
+    sp.setBoostStrength(0);
+
+    vector<UInt> input = { 1, 1, 0, 0, 1, 1, 0, 0, 1, 1 };
+    vector<UInt> out1(nColumns, 0);
+    vector<UInt> out2(nColumns, 0);
+    sp.compute(input.data(), false, out1.data());
+    sp.compute(input.data(), false, out2.data());
```

Comment:
* we'll definitely need more then one iteration, the issue seems to manifest "after some time" (-> an overflow?); I'm suggesting to test on a sequence of `random with a fixed seed`
* I'm not sure what @scottpurdy was thinking, I'd either compare two c++ SP instances, or better a c++ and py SP.

Reply:
I take it back. Neither of that is needed, I can replicate the issue now and will post the test-case. Seems to be related to initial params (and initialization via constructor vs. `sp.setXXX()`)